seat.c

```diff
@@ -192,7 +200,15 @@ handle_new_pointer(struct cg_seat *seat, struct wlr_input_device *device)
 	pointer->destroy.notify = handle_pointer_destroy;
 	wl_signal_add(&device->events.destroy, &pointer->destroy);
 
-	wlr_cursor_map_input_to_output(seat->cursor, device, seat->server->output->wlr_output);
+	if (device->output_name != NULL) {
+		struct cg_output *output;
+		wl_list_for_each(output, &seat->server->outputs, link) {
+			if (strcmp(device->output_name, output->wlr_output->name) == 0) {
+				wlr_cursor_map_input_to_output(seat->cursor, device, output->wlr_output);
```

Comment:
Can a pointer device really be bound to a single output? For a touchscreen this makes sense, but a pointer device?

Reply:
Yes, pointer devices can be relative or absolute, and in the latter case usually maps to an output. Drawing tablets are a good example of a physical absolute input device.

Most virtual pointers also happen to be absolute. This including the wlroots Wayland backend, whose output-specific pointers are completely bonkers if not mapped to their respective outputs.

Reply:
Thanks, that makes sense!